src/Illuminate/Foundation/Application.php

```diff
@@ -204,6 +205,21 @@ public function __construct($basePath = null)
         $this->registerCoreContainerAliases();
     }
 
+    /**
+     * Create a new Laravel application instance.
+     *
+     * @param  string|null  $baseDirectory
+     * @return static
+     */
+    public static function create(string $baseDirectory = null)
+    {
+        $baseDirectory = $ENV['APP_BASE_PATH'] ?? ($baseDirectory ?: dirname(dirname(
+            debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 1)[0]['file']
+        )));
```

Comment:
Is that intended to have the value from env override anything else?

It would seem unexpected if the calls to `Application::create($explicitPath)` ended up not using the provided argument.